.github/workflows/main.yml

```diff
@@ -97,6 +97,8 @@ jobs:
       run: echo CARGO_CONTAINER_TESTS=1 >> $GITHUB_ENV
       if: matrix.os == 'ubuntu-latest'
 
+    - name: Check if Cargo.lock is up-to-date
+      run: cargo fetch --locked
```

Comment:
The downside with doing this in a tet job is that it means someone doesn't collect any feedback if this error occurs.  This is also why distinct clippy jobs are nice so warnings about builds don't prevent seeing test errors or the other way around

Reply:
That's fair. Moved it to an independent job. I think it's fine keeping things going for now, but maintainers should keep an eye on lockfile being updated.

As a result of being a separate job, it uses more resource and energy, which is not really great 😆.

Reply:
Being a 13s job, I'm not too worried.  If this was more in the minutes, then I would be.